syntax = "proto3";

message CBSDStateRequest {
  string user_id = 1;
  string fcc_id = 2;
  string serial_number = 3;
}

message CBSDStateResult {
  optional LteChannel channel = 1;
  bool radio_enabled = 2;
}

message LteChannel {
  int64 low_frequency_hz = 1;
  int64 high_frequency_hz = 2;
  float max_eirp_dbm_mhz = 3;
}

service DPService {
    rpc GetCBSDState (CBSDStateRequest) returns (CBSDStateResult) {}
}
